subprojects/language-native/src/main/java/org/gradle/language/swift/plugins/SwiftBasePlugin.java

```diff
@@ -253,4 +259,18 @@ private StripSymbols extractAndStripSymbols(AbstractLinkTask link, Names names,
 
         return stripSymbols;
     }
+
+    private static class SwiftCppUsageCompatibilityRule implements AttributeCompatibilityRule<Usage> {
+        @Inject
+        public SwiftCppUsageCompatibilityRule() {
+        }
```

Comment:
nit, as another way (like `JavaBasePlugin` does it), you could make the class default scoped and remove this constructor.

Reply:
Done.